agent/engine/dependencygraph/graph.go

```diff
@@ -256,26 +264,15 @@ func verifyResourceDependenciesResolved(target *apicontainer.Container, existing
 func containerOrderingDependenciesCanResolve(target *apicontainer.Container,
 	dependsOnContainer *apicontainer.Container,
 	dependsOnStatus string) bool {
+
 	targetDesiredStatus := target.GetDesiredStatus()
 	dependsOnContainerDesiredStatus := dependsOnContainer.GetDesiredStatus()
 
 	switch dependsOnStatus {
+	case StartCondition:
+		return verifyContainerOrderingStatus(dependsOnContainer)
 
-	case "START":
-		// The 'target' container doesn't desire to move to either 'Created' or the 'steady' state,
-		// which is not allowed
-		if targetDesiredStatus != apicontainerstatus.ContainerCreated && targetDesiredStatus != target.GetSteadyStateStatus() {
```

Comment:
for my understanding, why was this check removed?

Reply:
I removed this because this function is getting called from verifyContainerOrderingStatusResolvable. It has a line which already checks this. https://github.com/ubhattacharjya/amazon-ecs-agent/blob/RefactorDependencyCode/agent/engine/dependencygraph/graph.go#L195